runtime/mrc/src/lib.rs

```diff
@@ -695,14 +694,8 @@ impl_runtime_apis! {
 			Ok(Msa::get_owner_of(&key))
 		}
 
-		fn check_delegations(delegator_msa_ids: Vec<MessageSenderId>, provider_msa_id: MessageSenderId) -> Result<BTreeMap<MessageSenderId, bool>, DispatchError> {
-			let mut map = BTreeMap::new();
-			let provider = pallet_msa::types::Provider(provider_msa_id);
-			for id in delegator_msa_ids {
-				let delegator = pallet_msa::types::Delegator(id);
-				map.insert(id, Msa::get_provider_info_of(provider, delegator).is_some());
-			}
-			Ok(map)
+		fn has_delegation(delegator: Delegator, provider: Provider) -> Result<bool, DispatchError> {
+			Ok(Msa::get_provider_info_of(provider, delegator).is_some())
```

Comment:
Question:
So will this always return true or false since is_some will evaluate only those two
Not sure you need a Result here then as Option will do, give you are Oking and no erring but maybe worth a try running a local test to throw an error from get_provider_info_of and see what happens

Reply:
If not you should do a match statement here and return error too

Reply:
I believe that this method needs to return a `Result` to conform to substrate runtime constraints. I believe we paired on this, no?

Reply:
Rational behind return an error is that false could happen due to delegator not authorizing provider or false could also happen that they don't even exists

Hence my point about match check and return a error if not found

Reply:
I'm not sure I follow. What are you trying to find out?

A couple of things we should keep in mind here:

1. All methods declared as runtime api methods must return `Result<_, DispatchError>`, so I'm not sure we can return a boolean here. I agree that since we are returning an `Ok` regardless, the Result seems superfluous. However, there is a chance that substrate reserves the error case here for issues dispatching calls to the runtime (that may be why the `Result<_, DispatchError>` is required.
2. The `get_provider_info_of` method is defined via macro so I'm not sure I can manipulate it to throw an error. There may be a way to mock it, but I'm not sure what insight we're trying to get here.

Reply:
I don't think you need to mock it 

I think I can now boil down to simple question


But I think we are good I checked the actual rpc and there is an error returned there given that was checked in before 


Approving it

Reply:
https://github.com/LibertyDSNP/mrc/blob/81e290c474cd12438ff37d0cebf563eb980cfd0e/pallets/msa/src/rpc/src/lib.rs#L82

This is what i was searching so we are good